File: ios/ConfigOptions.swift
//
//  ConfigOptions.swift
//  ExpoBoxoSdk
//
//  Created by Azamat Kushmanov on 18/9/24.
//

import ExpoModulesCore

struct ConfigOptions: Record {
    @Field
    var clientId: String = ""
    
    @Field
    var userId: String = ""
    
    @Field
    var language: String = "en"
    
    @Field
    var sandboxMode: Bool = false
    
    @Field
    var multitaskMode: Bool = false
    
    @Field
    var theme: String = "system"
    
    @Field
    var isDebug: Bool = false
    
    @Field
    var showPermissionsPage: Bool = true
    
    @Field
    var showClearCache: Bool = true
    
    @Field
    var showAboutPage: Bool = true
}

File: android/src/main/java/io/boxo/expo/ConfigOptions.kt
package io.boxo.expo

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class ConfigOptions : Record {
    @Field
    val clientId: String = ""

    @Field
    val userId: String = ""

    @Field
    val language: String = "en"

    @Field
    val sandboxMode: Boolean = false

    @Field
    val multitaskMode: Boolean = false

    @Field
    val theme: String = "system"

    @Field
    val isDebug: Boolean = false

    @Field
    val showPermissionsPage: Boolean = true

    @Field
    val showClearCache: Boolean = true

    @Field
    val showAboutPage: Boolean = true

    @Field
    val consentScreenConfig: ConsentScreenConfigOptions? = null
}